src/functions/template-tags/html.php

```diff
@@ -132,3 +132,85 @@ function tribe_disabled( $disabled, $echo = true ) {
 		}
 	}
 }
+
+/**
+ * Generates a string for the tribe-dependency attributes.
+ *
+ * @since TBD
+ *
+ * @param array<string,mixed> $deps       The passed array of dependencies.
+ *
+ * @return string             $dependency The string of dependencies attributes to add to the input.
+ */
+function tribe_format_field_dependency( $deps ) {
+	// Sanity check.
+	if ( empty( $deps ) ) {
+		return '';
+	}
+
+	// Let's be case-insensitive!
+	$deps = array_combine( array_map( 'strtolower', array_keys( $deps ) ), $deps );
+
+	// No ID to hook to? Bail.
+	if ( empty( $deps['id'] ) ) {
+		return;
+	}
+
+	$dependency = '';
+
+	$accepted = [
+		'id',
+		'is',
+		'is-not',
+		'is-empty',
+		'is-not-empty',
+		'is-numeric',
+		'is-not-numeric',
+		'is-checked',
+		'is-not-checked',
+	];
+
+	$valid_deps = array_intersect_key( $deps, array_flip( $accepted ) );
+
+	foreach ( $valid_deps as $attr => $value ) {
+		// Attributes are always lower case.
+		$attr = strtolower( $attr );
+
+		// Handle the ID component.
+		if ( 'id' === $attr ) {
+			// "Missing" quotes is intentional - they are added automagically and we want to avoid doubles.
+			$dependency .= " data-depends={$value}";
+			continue;
+		}
+
+		// Handle boolean values.
+		if ( is_bool( $value ) ) {
+			if ( $value ) {
+				$dependency .= " data-{$attr}";
+			} else {
+				if ( 0 === stripos( $attr, 'is-not-' ) ) {
+					$attr = str_replace( 'is-not-', 'is-', $attr );
+				} else {
+					$attr = str_replace( 'is-', 'is-not-', $attr );
+				}
+
+				$dependency .= " data-{$attr}";
+				continue;
+			}
+
+			continue;
+		}
+
+		// Handle string and "empty" values
+		// "Missing" quotes is intentional - they are added automagically and we want to avoid doubles.
```

Comment:
```suggestion
		/*
		 * Handle string and "empty" values.
		 *
		 * "Missing" quotes is intentional - they are added automagically and we want to avoid doubles.
		 */
```

Reply:
Are we 100% sure here, especially with escaping and values that may include spaces?

Reply:
Let me fiddle with it a bit more, but the dependencies were getting double-quoted between here and the template output. This was my solution for that.

Reply:
Think I've found a better solution...?

Reply:
✅